itenium-socks/src/app/socks/shop.component.ts

```diff
@@ -1,21 +1,44 @@
-import { Component } from '@angular/core';
+import {Component, OnInit} from '@angular/core';
 import { SocksService } from './socks.service';
-import { Observable } from 'rxjs';
+import {last, Observable} from 'rxjs';
 import { Sock } from './sock.model';
 import { AsyncPipe, NgFor } from '@angular/common';
+import {MatPaginator, PageEvent} from "@angular/material/paginator";
 
 @Component({
   selector: 'app-shop',
   standalone: true,
-  imports: [NgFor, AsyncPipe],
+  imports: [NgFor, AsyncPipe, MatPaginator],
   templateUrl: './shop.component.html'
 })
-export class ShopComponent {
+export class ShopComponent implements OnInit{
+  totalItems = 100;
+  pageSize = 8;
+  currentPage = 0;
   socks$!: Observable<Sock[]>;
+  socks: Sock[] = [];
 
   constructor(private socksService: SocksService) {}
 
   ngOnInit(): void {
     this.socks$ = this.socksService.get();
+    this.socks$.pipe(last())
+      .subscribe(allSocks => {
+        this.socks = this.getSocksPage(allSocks, this.currentPage, this.pageSize)
+        this.totalItems = allSocks.length
+      })
+  }
+
+  pageChanged(event: PageEvent) {
+    this.currentPage = event.pageIndex;
+    this.socks$.pipe(last())
+      .subscribe(allSocks => {
+        this.socks = this.getSocksPage(allSocks, this.currentPage, this.pageSize)
+      })
```

Comment:
Omwille van de `.subscribe` introduceer je namelijk subtiele fouten

Na een page change ga je hier opnieuw de http call uitvoeren.
Als er sindsdien een sock toegevoegd is, dan is de totalCount daarna incorrect!